core/tests/puppeteer-acceptance-tests/utilities/user/super-admin.ts

```diff
@@ -99,6 +176,203 @@ export class SuperAdmin extends BaseUser {
+    role = role.replace(/\b\w/g, char => char.toUpperCase());
+    await this.clickOn(role);
+  }
+
+  /**
+   * This function checks if the allocated actions for a role are present on the page.
+   * @param {string[]} actions - The actions to check for.
+   */
+  async expectRoleToHaveAllocatedActions(actions: string[]): Promise<void> {
+    for (const action of actions) {
+      const isActionPresent = await this.isTextPresentOnPage(action);
+      if (!isActionPresent) {
+        throw new Error(`Action ${action} is not allocated to the role`);
+      }
+    }
+    showMessage(`${actions} is/are allocated to the role`);
+  }
+
+  /**
+   * Checks if the specified users are assigned to the current role.
+   * @param {string[]} users - An array of usernames to check.
+   */
+  async expectRoleToHaveAssignedUsers(users: string[]): Promise<void> {
+    await this.clickOn(' Assigned users ');
+    await this.page.waitForTimeout(2000);
+    for (const user of users) {
+      const isUserPresent = await this.isTextPresentOnPage(user);
+      if (!isUserPresent) {
+        throw new Error(`User ${user} is not assigned to the role`);
+      }
+    }
+    showMessage(`${users} is/are assigned to the role`);
+  }
+
+  /**
+   * Checks if the application is in development mode.
+   * @returns {Promise<boolean>} Returns true if the application is in development mode, false otherwise.
+   */
+  async isInDevMode(): Promise<boolean> {
+    const prodMode = process.env.PROD_MODE;
+    return prodMode !== 'true';
+  }
+
+  async reloadExplorations(explorationName: string): Promise<void> {
+    await this.page.waitForSelector(reloadExplorationRowsSelector);
+    const reloadExplorationRows = await this.page.$$(
+      reloadExplorationRowsSelector
+    );
+    for (let i = 0; i < reloadExplorationRows.length; i++) {
+      const explorationNameElement = await reloadExplorationRows[i].$(
+        '.e2e-test-reload-exploration-title'
+      );
+      const name = await this.page.evaluate(
+        element => element.innerText,
+        explorationNameElement
+      );
+      if (name === ` ${explorationName} `) {
+        const reloadButton = await reloadExplorationRows[i].$(
+          '.e2e-test-reload-exploration-button'
+        );
+        if (reloadButton) {
+          await reloadButton.evaluate(element =>
+            (element as HTMLElement).click()
+          );
+          await this.page.waitForNetworkIdle();
+          return;
+        }
+      }
+    }
+  }
+
+  async expectExplorationToBePresent(explorationName: string): Promise<void> {
+    await this.type('input.e2e-test-search-input', explorationName);
+    const isExplorationPresent =
+      await this.isTextPresentOnPage(explorationName);
+    if (!isExplorationPresent) {
+      throw new Error(`Exploration ${explorationName} is not present`);
+    }
+  }
+
+  async navigateToCommunityLibrary(): Promise<void> {
+    await this.clickOn(CommunityLibraryUrl);
```

Comment:
Why did you click on a URL instead of using goto?

Reply:
updated.